components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1405,13 +1437,39 @@ void KeyringService::Lock() {
   StopAutoLockTimer();
 }
 
-bool KeyringService::IsHardwareAccount(const std::string& account) const {
-  for (const auto& hardware_account_info : GetHardwareAccountsSync()) {
-    if (hardware_account_info->address == account) {
-      return true;
+std::string KeyringService::GetKeyringIdForAccount(
+    const std::string& address) const {
+  auto keyrings = GetAvailableKeyringsFromPrefs(prefs_);
+  for (const auto& keyring : keyrings) {
+    std::vector<mojom::AccountInfoPtr> infos =
+        GetAccountInfosForKeyring(keyring);
+
+    // Check for matching default and imported account
+    for (const mojom::AccountInfoPtr& info : infos) {
+      if (base::EqualsCaseInsensitiveASCII(info->address, address)) {
+        return keyring;
+      }
     }
   }
-  return false;
+  return std::string();
+}
+
+std::string KeyringService::GetKeyringIdForHardwareAccount(
+    const std::string& address) const {
```

Comment:
This function is almost identical with `GetKeyringIdForAccount` expect for one is calling `GetAccountInfosForKeyring` to get account infos and one is calling `GetHardwareAccountsSync` to get account infos, you should combine this two to avoid code duplication.

Reply:
`GetKeyringIdForAccount` searches through all accounts, `GetKeyringIdForHardwareAccount` only by hardware accounts and cover different use cases in which we do not need any additional info except keyring id. how do you prefer to combine them?

Reply:
Added coin to `SetSelectedAccount` and removed `GetKeyringIdForAccount`